### ISSUE TITLE
Allow/document creation of custom pickup point providers

### ISSUE THREAD

**Issue by rimas-kudelis** [Setono/SyliusPickupPointPlugin#46]
The plugin only seems to support the few hardcoded pickup point providers, but the list of obviously incomplete, plus, it realistically cannot cover all possible scenarios. For example, I'm considering this plugin in a scenario where I'll be getting a list of pickup points from another internal system on a daily basis. Such functionality wouldn't make any sense as a PR to this project, but there doesn't seem to be any documentation on how this could be implemented.

**Comment by loevgaard** [comment]
Are you going to contact the internal system in the checkout process or are you planning to cache the pickup points? Both things are possible right now I guess :)

But I agree that we need better docs. We always do, but copying one of the other provider should give you some pointers :)

**Comment by rimas-kudelis** [comment]
After looking over the source code, I ended up implementing our own cached pickup point provider, which stores pickup points in the current pickup point cache table, and doesn't attempt to load them from external system on each request.

We're pulling them from another internal system nightly, which, in turn, pulls them from the provider (DPD) just a little before that.

I changed the title of this bug, because I figured out it's quite easily possible, just not documented.
